notification/src/api/notification/notification.controller.js

```diff
@@ -0,0 +1,133 @@
+const _ = require('lodash')
+const Promise = require('bluebird')
+const httpUtils = require('../../utils/commons')
+const config = require('../../config/config').load()
+const ctp = require('../../utils/ctp')
+const ctpClient = ctp.get(config)
+
+const adyenEvents = require('../../../resources/adyenEvents')
+
+// TODO: add JSON schema validation:
+// https://github.com/commercetools/commercetools-adyen-integration/issues/9
+async function handleNotification (request, response, logger) {
+  const body = await httpUtils.collectRequestData(request, response)
+  try {
+    await processNotifications(JSON.parse(body), logger)
+    response.statusCode = 200
+    return response.end("{ notificationResponse : '[accepted]' }")
+  } catch(e) {
+    logger.error(e, 'Ooops')
+    response.statusCode = 500
+    return response.end()
+  }
+}
+
+// TODO: proper naming
+async function processNotifications (request, logger) {
+  const notifications = request.notificationItems
+  await Promise.map(notifications, notification => processNotification(notification, logger))
+}
+
+// TODO: proper naming
+async function processNotification (notification, logger) {
+  const merchantReference = _.get(notification, 'NotificationRequestItem.merchantReference', null)
+  if (merchantReference === null) {
+    logger.error(`Can't extract merchantReference from the notification: ${JSON.stringify(notification)}`)
+    return null
+  }
+
+  let updateActions
+  let payment
+  try {
+    payment = await getPaymentByMerchantReference(merchantReference)
+    updateActions = calculateUpdateActionsForPayment(payment, notification)
+  } catch (e) {
+    logger.warn(e, `Error while trying to fetch the payment for merchantReference: ${merchantReference}`)
+    return null
+  }
+
+  try {
+    await ctpClient.update(ctpClient.builder.payments, payment.id, payment.version, updateActions)
+  } catch (e) {
+    logger.error(e, `Error while updating the payment with id: ${payment.id}`)
+  }
+}
+
+function calculateUpdateActionsForPayment (payment, notification) {
+  const notificationEventCode = notification.eventCode
+  const notificationSuccess = notification.success
+  const updateActions = []
+  const stringifiedNotification = JSON.stringify(notification)
+  // check if interfaceInteraction already on payment or not
```

Comment:
`interfaceInteraction` is always on payment object: https://docs.commercetools.com/http-api-projects-payments#payment

Reply:
https://github.com/commercetools/commercetools-adyen-integration/commit/9f3ca241cbff3f263973c82fd8e4f57f93a51950